e2e/imgbuild/imgbuild.go

```diff
@@ -154,6 +159,511 @@ func badPath(t *testing.T) {
 	}
 }
 
+func buildMultiStageDefinition(t *testing.T) {
+	tmpfile, err := ioutil.TempFile(testenv.TestDir, "testFile-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(tmpfile.Name()) // clean up
+
+	if _, err := tmpfile.Write([]byte(testFileContent)); err != nil {
+		log.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		log.Fatal(err)
+	}
+
+	tests := []struct {
```

Comment:
These test cases are really difficult to discern what is going on. I know that you're trying to avoid repetition with templates but perhaps here you should use actual definition files that people can view under /examples or /e2e/testdata ? What do you think?

Reply:
I've been thinking about this for a bit and I'm a bit torn. Though I agree this isn't great to read, and for these tests using an actual definition file would cut down the amount of information needed in this test struct, it would be a significant change from `buildDefinition` and potentially be confusing for someone to understand the difference between the two at first glance. Maybe it might be easier for a reader if these tests appeared after `buildDefinition` so that they're familiar with what that is doing and seeing what this is doing will be more intuitive?

Reply:
I just wanted to have a discussion here. As long as you've thought about it a bit and are open to more conversations I think it's something we can approach later.